source.lua

```diff
@@ -1,38 +1,68 @@
 --@ module = true
 local repeatUtil = require('repeat-util')
+local json = require('json')
+local persist = require('persist-table')
 
-liquidSources = liquidSources or {}
+local GLOBAL_KEY = 'source' -- used for state change hooks and persistence
 
 local sourceId = 'liquidSources'
 
+g_sources_list = g_sources_list or {}
+local function retrieve_state()
+    return g_sources_list
+end
```

Comment:
unless you're planning on writing unit tests, you don't need this extra level of indirection. you can use `g_sourcess_list` directly.